### PR TITLE
Update the transformers text example

### PR DESCRIPTION
**Pull request by zechengz**
Update the transformers text example with
- Fix the data save error when model name contains `/`
- Add mixed precision training, which saves memory and makes the training faster. But unfortunately for some datasets, marked as `N/A` in the comments, may need to tune or adjust the `scaler` scale or clip the gradient norm (https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_norm_.html) to get reasonable results
- Count and print number of finetune parameters for the text encoder.

### CODE REVIEW

File: examples/transformers_text.py [hunk 1]
```diff
@@ -54,6 +72,7 @@
 parser.add_argument('--seed', type=int, default=0)
 parser.add_argument('--finetune', action='store_true')
 parser.add_argument('--lora', action='store_true')
+parser.add_argument('--mixed', action='store_true')
```
Comment (weihua916) [inline examples/transformers_text.py:75]: rename to `mixed_precision_training`

**Review by weihua916 — APPROVED** [review]

**Comment by weihua916** [comment]
Personally, I'd prefer mixed precision not to be merged into `master` to avoid clutter on our example script. I am more comfortable with merging the first and third points in this PR, and keep experimenting with mixed precision training until we are confident that it works.

**Comment by zechengz** [comment]
@weihua916 Sure!

**Comment by akihironitta** [comment]
@zechengz In addition to `bf16`, you could also try disabling casting in particular operations that lead to the instability issue.
https://discuss.pytorch.org/t/disabling-mixed-precsion-in-my-own-layers/97190

**Comment by zechengz** [comment]
@akihironitta Thanks, I will have a try later.
